src/vmspawn/vmspawn-util.c

```diff
@@ -0,0 +1,186 @@
+                if (access(firmware, F_OK) == 0) {
+                        if (ret_firmware_path)
+                                *ret_firmware_path = firmware;
+                        return 0;
+                }
+        }
+
+#undef FIRMWARE_LOCATIONS
+        #define FIRMWARE_LOCATIONS \
+                "/usr/share/edk2/ovmf/OVMF_CODE.fd", \
+                "/usr/share/edk2-ovmf/OVMF_CODE.fd", \
+                "/usr/share/qemu/OVMF_CODE.fd", \
+                "/usr/share/ovmf/OVMF.fd", \
+                "/usr/share/OVMF/OVMF_CODE.fd"
+
+        FOREACH_STRING(firmware, FIRMWARE_LOCATIONS) {
+                if (access(firmware, F_OK) == 0) {
+                        if (ret_firmware_path)
+                                *ret_firmware_path = firmware;
+                        return 1;
+                }
+        }
+
+#undef FIRMWARE_LOCATIONS
+        return -ENOENT;
+}
+
+int find_qemu_binary(char **ret_qemu_binary) {
+        int r;
+
+        /*
+         * On success the path to the qemu binary will be stored in `req_qemu_binary`
+         *
+         * If the qemu binary cannot be found -ENOENT will be returned.
+         * If the native architecture is not supported by qemu -ESRCH will be returned;
+         */
+
+        static const char *architecture_to_qemu_table[_ARCHITECTURE_MAX] = {
+                [ARCHITECTURE_ARM64]       = "aarch64",     /* differs from our name */
+                [ARCHITECTURE_ARM]         = "arm",
+                [ARCHITECTURE_ALPHA]       = "alpha",
+                [ARCHITECTURE_X86_64]      = "x86_64",      /* differs from our name */
+                [ARCHITECTURE_X86]         = "i386",        /* differs from our name */
+                [ARCHITECTURE_LOONGARCH64] = "loongarch64",
+                [ARCHITECTURE_MIPS64_LE]   = "mips",        /* differs from our name */
+                [ARCHITECTURE_MIPS_LE]     = "mips",        /* differs from our name */
+                [ARCHITECTURE_PARISC]      = "hppa",        /* differs from our name */
+                [ARCHITECTURE_PPC64_LE]    = "ppc",         /* differs from our name */
+                [ARCHITECTURE_PPC64]       = "ppc",         /* differs from our name */
+                [ARCHITECTURE_PPC]         = "ppc",
+                [ARCHITECTURE_RISCV32]     = "riscv32",
+                [ARCHITECTURE_RISCV64]     = "riscv64",
+                [ARCHITECTURE_S390X]       = "s390x",
+        };
+
+        FOREACH_STRING(s, "qemu", "qemu-kvm") {
+                r = find_executable(s, ret_qemu_binary);
+                if (r == 0)
+                        return 0;
+
+                if (r != -ENOENT)
+                        return r;
+        }
+
+        const char *arch_qemu = architecture_to_qemu_table[native_architecture()];
+        if (!arch_qemu) {
+                return log_debug_errno(SYNTHETIC_ERRNO(ESRCH), "Architecture %s not supported by qemu", architecture_to_string(native_architecture()));
+        }
```

Comment:
coding style, no {} around single line if blocks.

Reply:
Fixed in: bbd52c2cd9